plugins/modules/wafv2_web_acl.py

```diff
@@ -213,22 +280,28 @@ def __init__(self, wafv2, name, scope, fail_json_aws):
         self.fail_json_aws = fail_json_aws
         self.existing_acl, self.id, self.locktoken = self.get_web_acl()
 
-    def update(self, default_action, description, rules, sampled_requests, cloudwatch_metrics, metric_name):
+    def update(self, default_action, description, rules, sampled_requests, cloudwatch_metrics, metric_name, custom_response_bodies):
+        req_obj = {
+            'Name': self.name,
+            'Scope': self.scope,
+            'Id': self.id,
+            'DefaultAction': default_action,
+            'Description': description,
+            'Rules': rules,
+            'VisibilityConfig': {
+                'SampledRequestsEnabled': sampled_requests,
+                'CloudWatchMetricsEnabled': cloudwatch_metrics,
+                'MetricName': metric_name
+            },
+            'LockToken': self.locktoken
+        }
+
+        if len(custom_response_bodies) > 0:
+            module.require_botocore_at_least('1.21.0', reason='to set custom response bodies')
```

Comment:
`module` is not available in the `update()` method.

Reply:
I guess I have to restructure my code a little then. Will look into this the coming days, as I don't have the bandwidth for it right now.

Reply:
It's pretty common to add this clause into main, and run the test prior to even creating your client.

(well it was before I got to delete most of the checks because we now have a minimum supported version)

Reply:
```diff
diff --git a/plugins/modules/wafv2_web_acl.py b/plugins/modules/wafv2_web_acl.py
index afc2161..3a5183c 100644
--- a/plugins/modules/wafv2_web_acl.py
+++ b/plugins/modules/wafv2_web_acl.py
@@ -273,11 +273,12 @@ from ansible_collections.community.aws.plugins.module_utils.wafv2 import wafv2_s
 
 
 class WebACL:
-    def __init__(self, wafv2, name, scope, fail_json_aws):
+    def __init__(self, wafv2, name, scope, fail_json_aws, require_botocore_at_least):
         self.wafv2 = wafv2
         self.name = name
         self.scope = scope
         self.fail_json_aws = fail_json_aws
+        self.require_botocore_at_least = require_botocore_at_least
         self.existing_acl, self.id, self.locktoken = self.get_web_acl()
 
     def update(self, default_action, description, rules, sampled_requests, cloudwatch_metrics, metric_name, custom_response_bodies):
@@ -297,7 +298,7 @@ class WebACL:
         }
 
         if len(custom_response_bodies) > 0:
-            module.require_botocore_at_least('1.21.0', reason='to set custom response bodies')
+            self.require_botocore_at_least('1.21.0', reason='to set custom response bodies')
             req_obj['CustomResponseBodies'] = custom_response_bodies
 
         try:
@@ -365,7 +366,7 @@ class WebACL:
         }
 
         if len(custom_response_bodies) > 0:
-            module.require_botocore_at_least('1.21.0', reason='to set custom response bodies')
+            self.require_botocore_at_least('1.21.0', reason='to set custom response bodies')
             req_obj['CustomResponseBodies'] = custom_response_bodies
         if description:
             req_obj['Description'] = description
@@ -434,7 +435,7 @@ def main():
     if not metric_name:
         metric_name = name
 
-    web_acl = WebACL(module.client('wafv2'), name, scope, module.fail_json_aws)
+    web_acl = WebACL(module.client('wafv2'), name, scope, module.fail_json_aws, module.require_botocore_at_least)
     change = False
     retval = {}
 
```

Reply:
I can commit to you branch (_if you like and if possible_). Sanity is passing with that.

Reply:
sure, feel free to make your own commits!

Reply:
I would however go with @trembles way of making the check centrally in `main`.